Assets/Scripts/Interaction/PlayerInteractor.cs

```diff
@@ -21,22 +22,50 @@ private void Start()
         {
             if (!isLocalPlayer) Destroy(this);
             hands = GetComponent<Hands>();
-            if (!mainCamera) mainCamera = Camera.main;
+            mainCamera = Camera.main;
         }
 
         protected virtual void Update()
         {
             if (!mainCamera) return;
+            if (!Input.GetButtonDown("Click")) return;
             
-            if (Input.GetButtonDown("Click"))
+            if (GetWorldData(out var hit))
             {
-                var ray = mainCamera.ScreenPointToRay(Input.mousePosition);
+                Interact(hit.transform, hit.point, hit.normal);
+            }
+        }
+
+        /// <summary>
+        /// Get the current interaction point data
+        /// </summary>
+        /// <param name="hit">Result of the raycast</param>
+        /// <returns></returns>
+        public bool GetWorldData(out RaycastHit hit)
+        {
+            var ray = mainCamera.ScreenPointToRay(Input.mousePosition);
+            //Ignore "Ignore Raycast" layer.
+            LayerMask layerMask = ~(1 << LayerMask.NameToLayer ("Ignore Raycast"));
+            bool hitSomething = Physics.Raycast(ray, out hit, float.PositiveInfinity, layerMask);
+            // Ensure that user did not click the UI
+            return !EventSystem.current.IsPointerOverGameObject() && hitSomething;
```

Comment:
Please check if the UI is being pointed at before doing the raycast, as one is clearly less expensive that the other.